tests/unit/components/report/setup.js

```diff
@@ -0,0 +1,34 @@
+
+function _hasPassed(test) {
+  return test.state === 'passed';
+}
+
+module.exports = function(data) {
+  let numberOfPasses = 0,
+      numberOfFailures = 0,
+      duration = 0,
+      suites = data;
+
+  suites.forEach(function(suite) {
+    suite.passes = suite.tests.filter(function(test) {
+      return _hasPassed(test);
+    });
+    numberOfPasses += suite.passes.length;
+
+    suite.failures = suite.tests.filter(function(test) {
```

Comment:
You're mutating the `suite` object here. Do you really need this property? If so, clone the object first.